sriovnet_switchdev.go

```diff
@@ -0,0 +1,92 @@
+package sriovnet
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	netdevPhysSwitchID = "phys_switch_id"
+	netdevPhysPortName = "phys_port_name"
+)
+
+var physPortRe = regexp.MustCompile(`pf(\d+)vf(\d+)`)
+
+func parsePortName(physPortName string) (pfRepIndex int, vfRepIndex int, err error) {
+
+	pfRepIndex = -1
+	vfRepIndex = -1
+
+	// old kernel syntax of phys_port_name is vf index
+	physPortName = strings.TrimSpace(physPortName)
+	physPortNameInt, err := strconv.Atoi(physPortName)
+	if err == nil {
+		vfRepIndex = physPortNameInt
+	} else {
+		// new kernel syntax of phys_port_name pfXVfY
+		matches := physPortRe.FindStringSubmatch(physPortName)
+		if len(matches) != 3 {
+			err = fmt.Errorf("Failed to parse physPortName %s", physPortName)
+		} else {
+			pfRepIndex, err = strconv.Atoi(matches[1])
+			if err == nil {
+				vfRepIndex, err = strconv.Atoi(matches[2])
+			}
+		}
+
+	}
+	return pfRepIndex, vfRepIndex, err
+}
+
+func GetVfRepresentor(uplink string, vfIndex int) (string, error) {
```

Comment:
needs comment to explain what is 'uplink' such as uplink netdev. Returns Vf representor netdev.
Lets improve next time.